### PR TITLE
Add Resource#wait_until regression with deprecate warnings

### PR DESCRIPTION
**Pull request by cjyclaire**
Address #1610 

Some background, in V2 those resource [objects](http://docs.aws.amazon.com/sdkforruby/api/Aws/EC2/Volume.html) all inherits [`Aws::Resources::Resource`](http://docs.aws.amazon.com/sdkforruby/api/Aws/Resources/Resource.html). This [`#wait_until`](https://github.com/aws/aws-sdk-ruby/blob/version-2/aws-sdk-resources/lib/aws-sdk-resources/resource.rb#L101) method looks like the only that make sense and need extra port over to V3. This PR ported this over to all resource objects that has waiters for their service, avoid depending on `resource.json` waiter check because `resource.json` usually quite outdated. Tested manually.

Other alternative approaches considered yet not followed:
I have considered port this over to` resource.rb`s solo, and let those objects' `#wait_until` just call the one of service resource. However this won't work for block usage [here](https://github.com/aws/aws-sdk-ruby/blob/version-2/aws-sdk-resources/lib/aws-sdk-resources/resource.rb#L102).

Notes, this usage behavior looks like should have be been [deprecated](https://github.com/aws/aws-sdk-ruby/blob/master/gems/aws-sdk-ec2/lib/aws-sdk-ec2/client.rb#L21323), so I have added @deprecated tag as well

Sample diff almost the same with the mustache diff : )

### CODE REVIEW

**Comment by cjyclaire** [comment]
Ah will follow up with tests

**Comment by awood45** [comment]
Looks good on a first pass, will wait for tests.

**Comment by cjyclaire** [comment]
Thought the same. I also thought S3 test might be cleaner, however S3 waiters might not be a good place to test states from block, so I mock the same EC2 issue here. I also make sure the volume is cleaned up. Now ready for another review : )

File: gems/aws-sdk-ec2/features/step_definitions.rb [hunk 1]
```diff
@@ -4,5 +4,24 @@
...
+  @volume_id = @volume.id
+end
+
+When(/^I use \#wait_until to wait until volume is available$/) do
+  expect {
+    @resp = @volume.wait_until {|v| v.state == 'available'}
+  }.not_to raise_error
+end
+
+Then(/^Waiter works and volume is cleaned up$/) do
```
Comment (awood45) [inline gems/aws-sdk-ec2/features/step_definitions.rb]: Use the cleanup logic you stubbed above. If the test fails you won't clean up the resource as written now.

Comment (cjyclaire) [reply gems/aws-sdk-ec2/features/step_definitions.rb]: Make sense, fixed in the latest commit.

**Pull request merged**